src/osw/express.py

```diff
@@ -350,55 +361,92 @@ class DataModel(OswBaseModel):
 
 
 def import_with_fallback(
-    to_import: List[DataModel], dependencies: Dict[str, str] = None, domain: str = None
+    to_import: Union[List[DataModel], Dict[str, str]],
+    caller_globals: dict,
+    module: str = None,
+    dependencies: Dict[str, str] = None,
+    domain: str = None,
 ):
     """Imports data models with a fallback to fetch the dependencies from an OSL
     instance if the data models are not available in the local osw.model.entity module.
 
     Parameters
     ----------
     to_import
-        List of DataModel objects to import.
+        List of DataModel objects or a dictionary, with (key: value) pairs (class_name:
+        osw_fpt) to import.
+    caller_globals
+        (Mandatory!) The globals dictionary as returned by globals() in the calling
+        script.
+    module
+        (Optional) The module to import the data models from. Used only if to_import
+        is of type List[Dict]. Defaults to 'osw.model.entity' if not specified.
     dependencies
         A dictionary with the keys being the names of the dependencies and the values
         being the full page name of the dependencies.
     domain
         The domain of the OSL instance to connect to, if the dependencies are not
         available in the local osw.model.entity module.
 
+    Examples
+    --------
+    >>> import_with_fallback(
+    >>>     [DataModel(module="osw.controller.file.base",class_name="FileController")],
+    >>>     globals(),
+    >>> )
+
     Returns
     -------
-
+    None
     """
+    if isinstance(to_import, dict):
+        # The arg 'to_import' should have the right structure to act as 'dependencies'
+        # Assume all DataModels are part of osw.model.entity
+        if module is None:
+            module = "osw.model.entity"
+        # A list of DataModels will be derived from the dict
+        to_import = [
+            DataModel(
+                module=module,
+                class_name=key,
+                osw_fpt=value,
+            )
+            for key, value in to_import.items()
+        ]
+    if to_import is None:
+        raise ValueError(
+            "Either the argument 'to_import' or 'dependencies' must be passed to the "
+            "function import_with_fallback()."
+        )
     try:
+        # Try to import the listed data models from the (listed) module(s)
         for ti in to_import:
             # Raises AttributeError if the target could not be found
-            globals()[ti.class_name] = getattr(
+            caller_globals[ti.class_name] = getattr(
```

Comment:
WIthout forcing the globals() from the most outer namespace to be passed to the function, the imported modules would only be available within osw.express namespace